src/wp-includes/blocks.php

```diff
@@ -151,10 +151,12 @@ function register_block_style_handle( $metadata, $field_name ) {
 		$style_uri  = includes_url( 'blocks/' . str_replace( 'core/', '', $metadata['name'] ) . "/style$suffix.css" );
 	}
 
-	$style_handle = generate_block_asset_handle( $metadata['name'], $field_name );
-	$block_dir    = dirname( $metadata['file'] );
-	$style_file   = realpath( "$block_dir/$style_path" );
-	$version      = file_exists( $style_file ) ? filemtime( $style_file ) : false;
+	$style_handle   = generate_block_asset_handle( $metadata['name'], $field_name );
+	$block_dir      = dirname( $metadata['file'] );
+	$style_file     = realpath( "$block_dir/$style_path" );
+	$has_style_file = false !== $style_file;
+	$version        = ! $is_core_block && $has_style_file ? filemtime( $style_file ) : false;
+	$style_uri      = ( $is_core_block && $style_file ) || ! $is_core_block ? $style_uri : false;
```

Comment:
With this conditional, I don't think non-Core blocks would ever be set to `false` if the style file does not exist. Could this be shortened to `$style_file_exists ? $style_uri : false;`?

Reply:
```suggestion
	$style_uri      = $has_style_file ? $style_uri : false;
```

I tend to agree with @desrosj . If the file doesn't exist, then the URI would be `false`, meaning the URI is only set when the file exists.